app/ts/components/TransactionDetailsPage.tsx

```diff
@@ -0,0 +1,221 @@
+import { ComponentChildren } from 'preact'
+import { ethers } from 'ethers'
+import * as Layout from './Layout.js'
+import { useParams } from './HashRouter.js'
+import { assertUnreachable, calculateGasFee, isTransactionHash } from '../library/utilities.js'
+import { assertTransferStatus, transferStore } from '../store/transfer.js'
+import { useAsyncState } from '../library/preact-utilities.js'
+import * as Icon from "./Icon/index.js"
+
+export const TransactionDetailsPage = () => {
+	return (
+		<Layout.Page>
+			<Layout.Header />
+			<Layout.Body>
+				<PageTitle />
+				<Main />
+			</Layout.Body>
+			<Layout.Footer />
+		</Layout.Page>
+	)
+}
+
+const Main = () => {
+	const transfer = transferStore
+	const hash = extractTransactionHashFromParams()
+
+	if (!hash) return <TransactionDetailsInvalid />
+
+	switch (transfer.value.status) {
+		case 'idle':
+			return <TransactionDetailsIdle />
+		case 'signed':
+			return <TransactionDetailsSigned />
+		case 'confirmed':
+			return <TransactionDetailsConfirmed />
+		case 'new':
+			return null // render should be handled in send page
+		default: assertUnreachable(transfer.value)
+	}
+}
+
+const TransactionDetailsIdle = () => {
+	const transfer = transferStore.value
+	const hash = extractTransactionHashFromParams()
+	const [transactionResponse, resolveTransactionResponse, resetTransactionResponse] = useAsyncState()
+
+	assertTransferStatus(transfer.status, 'idle')
+
+	const fetchTransactionResponse = () => {
+		resolveTransactionResponse(async () => {
+			if (!hash || transferStore.value.status !== 'idle') return
+			await transferStore.value.fetchTransactionByHash(hash)
+		})
+	}
+
+	switch (transactionResponse.state) {
+		case 'inactive':
+			fetchTransactionResponse()
+			return null
```

Comment:
```suggestion
			return undefined
```

Reply:
these returns (including other suggestions) are from a functional component was very common to use `null` when you don't want to display or render visually on page. Preact would handle these the same anyways (null, boolean, undefined), but be nice if we have a standard one across projects. Thoughts @MicahZoltu ?

Reply:
I have a preference for `undefined` everywhere, never `null`.  This is just because there are so few situations where there is a meaningful difference between the two, and I think there is value in having consistency.

Reply:
I only use nulls when I have some function returning null and I cannot change that :). Eg `browser.windows.create` can return null

Reply:
a function returning undefined isn't valid JSX tho. it only works when wrapped with a fragment `<>{undefined}</>`

<img width="695" alt="Screenshot 2023-02-02 at 2 05 07 AM" src="https://user-images.githubusercontent.com/1169838/216126521-ab6fb3b2-42f3-409f-b97a-631129b22e2c.png">

Reply:
Huh, I didn't know that `null` was valid JSX but `undefined` isn't!  I will need to think on this and potentially alter my position on `null`s in preact projects.  🤔